### PR TITLE
fix(sonarr)!: deprecate v2

### LINKED ISSUE #596: Remove Sonarr v2
Sonarr v2 is EOL and now no longer supported

- [x]  Migrate all sonarr v2 to v3 (upgrade script exists)
- [x] Remove Sonarr v2 install script
- [x] determine what is happening with Sonarr v2 install script
- [x] determine if renaming Sonarrv3 scripts
- [ ] update docs

### PR DESCRIPTION
**Pull request by bakerboy448**
## Description

- Removes Sonarr v2 scripts
- Updates Sonarr v2 to v3  (can discuss if needed; if so then we keep the upgrade script and do not have  a sonarr update script)

<details>
<summary>Label of collapsible goes here</summary>

</details>

## Fixes issues: 
- Fixes #596

## Proposed Changes:
- see above

## Change Categories

- Breaking change 
- Other

## Checklist

- [x] Docs have been made OR are not necessary;  - need to discuss what's happening with docs
    - PR link: https://github.com/liaralabs/docs.swizzin.ltd/pull/68
- [x] Changes to panel have been made OR are not necessary - do we want to remove the legacy code yet?
    - PR link: https://github.com/liaralabs/swizzin_dashboard/pull/34
- [X] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling) - literal copy/paste of existing
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling) - hopefully
- [X] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [X] I have commented my code, particularly in hard-to-understand areas

## Test scenarios

if this doesn't work i'd be shocked.
### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|			|				|
| Bionic		|			|			|			|				|
| Buster		|			|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|	⚫️		|			|	⚫️		|	⚫️			|

### CODE REVIEW

**Comment by flying-sausages** [comment]
We have to make a call on renaming sonarrv3 to `sonarr`, and `sonarr` to `sonarr-oldv2` or something in the lockfiles so that people can still upgrade properly.

**Review by flying-sausages — CHANGES_REQUESTED** [review]
See comment above

**Comment by bakerboy448** [comment]
Should be set here.

We should let this populate before removing v2 from docs and panel.

**Comment by flying-sausages** [comment]
I'll test this when I get the chance

File: scripts/update/sonarr.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+fi
+if [[ -f /install/.sonarr.lock ]] && [[ $v2present == "true" ]]; then
+    #update lock file
+    rm /install/.sonarr.lock
+    touch /install/.sonarrv2-old.lock
+fi
+if [[ -f /install/.sonarrv2-old.lock ]]; then
+    #upgrade sonarr v2 to v3
+    box install sonarrv3
+fi
```
Comment (flying-sausages) [inline scripts/update/sonarr.sh:14]: This cannot be a forced migration. We have a manual migration for sonarr already, this is not going to pass. If someone chooses to stay on v2, so be it.

File: scripts/upgrade/sonarr.sh [hunk 2]
```diff
@@ -1,8 +0,0 @@
-#!/bin/bash
-
-if [[ ! -f /install/.sonarr.lock ]]; then
-    echo_error "Sonarr not detected. Exiting!"
-    exit 1
-fi
-
-box install sonarrv3
```
Comment (flying-sausages) [inline scripts/upgrade/sonarr.sh:8]: This upgrader should stay. It should instead check for `sonarrv2-old` and install `sonarr`

**Review by flying-sausages — CHANGES_REQUESTED** [review]
See comments above

File: scripts/update/sonarr.sh (hunk 1 above)
Comment (bakerboy448) [reply scripts/update/sonarr.sh]: no forced migration from v2 to v3 at this time

File: scripts/update/mono.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 #Have I mentioned I hate mono?
 
-if [[ -f /install/.sonarr.lock ]] || [[ -f /install/.radarr.lock ]] || [[ -f /install/.lidarr.lock ]]; then
+if [[ -f /install/.sonarr.lock ]]; then
```
Comment (flying-sausages) [inline scripts/update/mono.sh:4]: This should be reverted, we don't know when #343 will be merged

Comment (bakerboy448) [reply scripts/update/mono.sh:4]: not fully reverted, radarr still doesn't need mono

i did add lidarr back and added a comment on 343 to remove lidarr from this script at that time.

**Comment by flying-sausages** [comment]
If the lockfile names are being changed, we might need to adapt the panel for this btw

**Comment by bakerboy448** [comment]
https://github.com/liaralabs/swizzin_dashboard/search?q=sonarrv3.lock

Doesn't appear any panel changes are needed?

**Comment by flying-sausages** [comment]
> https://github.com/liaralabs/swizzin_dashboard/search?q=sonarrv3.lock
> 
> Doesn't appear any panel changes are needed?

the lockfiles themselves aren't hardcoded in the repo

https://github.com/liaralabs/swizzin_dashboard/search?q=sonarr

**Comment by bakerboy448** [comment]
comment still stands

I don't see any required changes?

eventually we should remove sonarr v2

**Comment by flying-sausages** [comment]
Yeah it's not blocking, just shouldn't be forgotten in the cleanup procedure

**Comment by flying-sausages** [comment]
Actually no hold up I take that back, it _is_ blocking due to the panel entries failing due to the change of how it expects the sonarr service to be called. I'll make a quick PR for it.

**Comment by flying-sausages** [comment]
Docs and Panel PRs added

**Comment by flying-sausages** [comment]
@liaralabs do the hyphens in the lockfiles create any issues?

**Comment by liaralabs** [comment]
Lock files cannot contain hyphens.

**Comment by bakerboy448** [comment]
hyphens cleared

**Comment by flying-sausages** [comment]
I only tested the `sonarrv3` to `sonarr` migration as I cannot install v2 sonarr anymore as there's no `nzbdrone` package anymore.

**Review by flying-sausages — APPROVED** [review]
ring me if it don't work but i'm confident in this

**Comment by liaralabs** [comment]
The remove script needs to have its filename updated to remove the hyphen as well:

```
 scripts/remove/sonarrv2-old.sh
 ```
 to
 ```
  scripts/remove/sonarrv2old.sh
  ```

Every single box function either checks for the presence or absence of the lockfile by the name of the script being called.

That said is there a reason why the name to call the remove script is so verbose? It looks especially awkward without a hyphen. Why are we not just using sonarrv2?

**Comment by bakerboy448** [comment]
hyphens removed

verbose simplified

**Comment by flying-sausages** [comment]
> hyphens removed
> 
> verbose simplified

Liara meant that you should rename the file, not change all the names of the lockfiles and stuff

**Comment by flying-sausages** [comment]
we can't have `sonarr` and `sonarrv2`. We need to find a different name to highlight that "v2" is old.

**Comment by liaralabs** [comment]
You are overthinking it. Sonarr v2 will no longer be installable, how will this cause confusion? The user will only ever be able to install `sonarr`. At this point we are essentially discussing what the remove script should be called and `sonarrv2` is appropriate.

I think calling it sonarrv2old is simply unnecessary.

File: scripts/update/mono.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 #Have I mentioned I hate mono?
 
-if [[ -f /install/.sonarr.lock ]] || [[ -f /install/.radarr.lock ]] || [[ -f /install/.lidarr.lock ]]; then
+if [[ -f /install/.sonarr.lock ]] || [[ -f /install/.lidarr.lock ]]; then
```
Comment (flying-sausages) [inline scripts/update/mono.sh:4]: sneaky sneaky

**Review by flying-sausages — APPROVED** [review]
tested and working

**Pull request merged**